### PR TITLE
github: add stale bot to dev docs repo

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Same action being used in other InstructLab repos

### CODE REVIEW

**Review by bjhargrave — APPROVED** [review]
You may want to con

File: .github/workflows/stale_bot.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  stale:
+    permissions:
+      issues: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Harden Runner"
+        uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
```
Comment (danmcp) [inline .github/workflows/stale_bot.yml:29]: Is this because we aren't sure if egress is needed yet?

Comment (nathan-weinberg) [reply .github/workflows/stale_bot.yml:29]: I believe so yes - @bjhargrave can you confirm?

Comment (bjhargrave) [inline .github/workflows/stale_bot.yml:29]:
```suggestion
          disable-sudo: true
          egress-policy: block
          allowed-endpoints: >
            api.github.com:443
```

For these stale workflows, we can configure the egress policy as above.

I should get around to making PRs to tighten this up for all the workflows in all the repos.

Comment (nathan-weinberg) [reply .github/workflows/stale_bot.yml:29]: Will make this change!

Comment (nathan-weinberg) [reply .github/workflows/stale_bot.yml]: Fixed in latest patchset

**Review by bjhargrave — APPROVED** [review]

**Review by danmcp — APPROVED** [review]

**Pull request merged**